### PR TITLE
Disable debug by default and use env TIDB_DEBUG=1 to enable

### PR DESCRIPTION
**Pull request by Unknwon**

### CODE REVIEW

File: tidb.go [hunk 1]
```diff
@@ -88,7 +89,7 @@ var (
 	}
 	stores = make(map[string]kv.Driver)
 	// Debug is the switch for pprof. If Debug is true, tidb will start with pprof on.
-	Debug = true
+	Debug = os.Getenv("TIDB_DEBUG") == "1"
```
Comment (ngaut) [inline tidb.go]: Please set Debug to true by default.

Comment (Unknwon) [inline tidb.go]: Even though I agree debug should be true by default, but I don't think start listening on a port is desired behavior of users. 

Maybe we should use `TIDB_PPROF=1`?

Comment (ngaut) [inline tidb.go]: I prefer TIDB_PPROF too.

Comment (Unknwon) [inline tidb.go]: Just to make sure we're on the same page, by default, should be `TIDB_PPROF=0`?

Comment (ngaut) [inline tidb.go]: Right now, TIDB_PPROF should be 1 by default.
We may set to 0 when we release beta version.

Comment (Unknwon) [inline tidb.go]: For `TIDB_PPROF`, should I create a new variable? 

If so, how does it affect `if` statement for starting listening?

Should it be `if Debug && EnablePprof` or just `if EnablePprof`?

Comment (ngaut) [inline tidb.go]: Just EnablePprof is fine.

**Comment by Unknwon** [comment]
Updated.

**Comment by ngaut** [comment]
Please pass go lint.

File: tidb.go [hunk 2]
```diff
@@ -88,7 +89,8 @@ var (
 	}
 	stores = make(map[string]kv.Driver)
 	// Debug is the switch for pprof. If Debug is true, tidb will start with pprof on.
-	Debug = true
+	Debug       = true
+	EnablePprof = os.Getenv("TIDB_PPROF") != "0"
 	// PprofAddr is the pprof url.
 	PprofAddr = "localhost:8888"
```
Comment (siddontang) [inline tidb.go]: should use env for pprof address too?

Comment (Unknwon) [inline tidb.go]: Maybe...

Comment (ngaut) [inline tidb.go]: As i mentioned in #684, we will use configuration to handle this.
Right now, let's just handle the flag.
@Unknwon CI failed, please add comment to pass go lint.  https://travis-ci.org/pingcap/tidb/builds/95520320

**Comment by Unknwon** [comment]
Updated.

**Comment by ngaut** [comment]
LGTM

**Comment by coocood** [comment]
LGTM

File: tidb.go [hunk 3]
```diff
@@ -89,6 +90,8 @@ var (
 	stores = make(map[string]kv.Driver)
 	// Debug is the switch for pprof. If Debug is true, tidb will start with pprof on.
 	Debug = true
```
Comment (shenli) [inline tidb.go]: Debug variable is useless now?

Comment (Unknwon) [inline tidb.go]: Currently, yes.

**Comment by shenli** [comment]
LGTM

**Pull request merged**